### PR TITLE
Add dockerhub authentication for kubelet (v7)

### PR DESCRIPTION
**Pull request by wtty-fool**
Towards giantswarm/giantswarm#13981

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: files/conf/kubelet-docker-config.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "auths": {
+        "https://index.docker.io/v1/": {
+            "auth": {{ .DockerhubToken }}
```
Comment (kopiczko) [inline files/conf/kubelet-docker-config.json:4]: Shouldn't be quoted?

```suggestion
            "auth": "{{ .DockerhubToken }}"
```

File: pkg/template/types.go [hunk 2]
```diff
@@ -25,8 +25,11 @@ type Params struct {
 	// the Ingress Controller service. This allows us to migrate providers to
 	// chart-operator independently.
 	DisableIngressControllerService bool
-	Etcd                            Etcd
-	Extension                       Extension
+	// DockerhubToken is an auth token used by kubelet to
+	// authenticate/authorize against https://index.docker.io/v1/.
+	DockerhubToken string
```
Comment (kopiczko) [inline pkg/template/types.go:30]: Don't we need to fail when it's empty (if so that would be a breaking change)?

Comment (wtty-fool) [reply pkg/template/types.go:30]: We need to fail 100%.

**Comment by wtty-fool** [comment]
Tested on KVM:

No credentials (I manually edited correct file created by `kvm-operator`):
```
kube-system   metrics-server-557fb87b85-kj54v                        0/1     ImagePullBackOff    0          44s    172.24.43.72     worker-mc2u0-64fd59dd56-tkc5k   <none>           <none> 
kube-system   net-exporter-jc7bp                                     0/1     Init:ErrImagePull   0          42s    <none>           worker-mc2u0-64fd59dd56-tkc5k   <none>           <none> 
```
```
k describe pods -n kube-system metrics-server-557fb87b85-kj54v | grep QPS
  Warning  Failed          14m                kubelet            Failed to pull image "quay.io/giantswarm/metrics-server-amd64:v0.3.6": pull QPS exceeded
```

With valid credentials (I restored the file to original state):
```
  Normal   Scheduled       14m                default-scheduler  Successfully assigned kube-system/metrics-server-557fb87b85-kj54v to worker-mc2u0-64fd59dd56-tkc5k
  Warning  Failed          14m                kubelet            Failed to pull image "quay.io/giantswarm/metrics-server-amd64:v0.3.6": pull QPS exceeded
  Warning  Failed          14m                kubelet            Error: ErrImagePull
  Normal   SandboxChanged  14m                kubelet            Pod sandbox changed, it will be killed and re-created.
  Normal   BackOff         14m (x2 over 14m)  kubelet            Back-off pulling image "quay.io/giantswarm/metrics-server-amd64:v0.3.6"
  Warning  Failed          14m (x2 over 14m)  kubelet            Error: ImagePullBackOff
  Normal   Pulling         14m (x2 over 14m)  kubelet            Pulling image "quay.io/giantswarm/metrics-server-amd64:v0.3.6"
  Normal   Pulling         13m                kubelet            Pulling image "quay.io/giantswarm/metrics-server-amd64:v0.3.6"
  Normal   Pulled          13m                kubelet            Successfully pulled image "quay.io/giantswarm/metrics-server-amd64:v0.3.6"
  Normal   Created         13m                kubelet            Created container metrics-server
  Normal   Started         13m                kubelet            Started container metrics-server

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+### Added
+
+- - Add dockerhub authentication for `kubelet`.
```
Comment (kopiczko) [inline CHANGELOG.md:12]:
```suggestion
- Add dockerhub authentication for `kubelet`.
```

**Review by kopiczko — APPROVED** [review]

**Pull request merged**